interfaces/builtin/repowerd.go

```diff
@@ -0,0 +1,307 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
+)
+
+var repowerdPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the repowerd service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# HW alarms
+/dev/alarm rw,
+
+# Suspend
+/sys/power/** rw,
+
+# Backlight
+/sys/**/brightness rw,
+/sys/**/max_brightness r,
+/sys/**/type r,
+
+# DBus
+#include <abstractions/dbus-strict>
+
+dbus (send)
+  bus=system
+  path=/org/freedesktop/DBus
+  interface=org.freedesktop.DBus
+  member={GetId,RequestName,ReleaseName,GetConnectionUnixProcessID}
+  peer=(name=org.freedesktop.DBus),
+
+dbus (receive, send)
+  bus=system
+  interface=org.freedesktop.DBus.{Properties,Introspectable},
+
+dbus (bind)
+  bus=system
+  name=com.canonical.powerd,
+
+dbus (bind)
+  bus=system
+  name=com.canonical.Unity.Screen,
+
+dbus (receive, send)
+  bus=system
+  interface=com.canonical.powerd,
+
+dbus (receive, send)
+  bus=system
+  interface=com.canonical.Unity.Screen,
+
+# logind session
+dbus (send)
+  bus=system
+  path=/org/freedesktop/login1
+  interface=org.freedesktop.login1.Manager
+  member=GetSessionByPID
+  peer=(name=org.freedesktop.login1),
+dbus (send)
+  bus=system
+  path=/org/freedesktop/login1{,/seat/**,/session/**}
+  interface=org.freedesktop.DBus.Properties
+  member=Get{,All}
+  peer=(name=org.freedesktop.login1),
+dbus (receive)
+  bus=system
+  path=/org/freedesktop/login1
+  interface=org.freedesktop.login1.Manager
+  member={SessionAdded,SessionRemoved}
+  peer=(name=org.freedesktop.login1),
+dbus (receive)
+  bus=system
+  path=/org/freedesktop/login1{,/seat/**,/session/**}
+  interface=org.freedesktop.DBus.Properties
+  member=PropertiesChanged
+  peer=(name=org.freedesktop.login1),
+
+# logind power
+dbus (send)
+  bus=system
+  path=/org/freedesktop/login1
+  interface=org.freedesktop.login1.Manager
+  member={Hibernate,HybridSleep,Inhibit,PowerOff,Suspend}
+  peer=(name=org.freedesktop.login1),
+
+# Unity.Display
+dbus (send)
+  bus=system
+  interface=com.canonical.Unity.Display
+  path=/com/canonical/Unity/Display
+  member={TurnOn,TurnOff}
+  peer=(name=com.canonical.Unity.Display),
```

Comment:
Is this part of the unity8 interface? If not, what is providing this service? If so, then the unity8 interface or some other unity8 interface should have ConnectedSlot and ConnectedPlug policy such that repowerd may `plugs: [ unity8 ]`. As written, it assumes that Unity.Display is around, but it doesn't exist in the core snap and therefore you need to connect this snap to the snap that provides this service. More concretely, we need a `peer=(label=...)` match but we don't have anything to set the label to.